vpr/src/route/rr_graph.cpp

```diff
@@ -1469,40 +1469,47 @@ static void build_rr_sinks_sources(const int i,
-                            L_rr_node[inode].set_type(IPIN);
+                                //Add info about the edge to be created
+                                rr_edges_to_create.emplace_back(inode, to_node, delayless_switch);
 
+                                VTR_ASSERT(inode >= 0);
+                                L_rr_node[inode].set_cost_index(IPIN_COST_INDEX);
+                                L_rr_node[inode].set_type(IPIN);
+                            }
                         } else {
                             VTR_ASSERT(class_inf[iclass].type == DRIVER);
                             //Initialize the output pin
                             // Note that we leave it's out-going edges unconnected (they will be hooked up to global routing later)
                             inode = get_rr_node_index(L_rr_node_indices, i + width_offset, j + height_offset, OPIN, ipin, side);
 
-                            //Initially left unconnected
-                            VTR_ASSERT(inode >= 0);
-                            L_rr_node[inode].set_cost_index(OPIN_COST_INDEX);
-                            L_rr_node[inode].set_type(OPIN);
+                            /* Output pins may not exist on some sides, we may not always find one */
+                            if (OPEN != inode) {
+                                //Initially left unconnected
+                                VTR_ASSERT(inode >= 0);
+                                L_rr_node[inode].set_cost_index(OPIN_COST_INDEX);
+                                L_rr_node[inode].set_type(OPIN);
+                            }
                         }
 
                         /* Common to both DRIVERs and RECEIVERs */
-                        L_rr_node[inode].set_capacity(1);
-                        float R = 0.;
-                        float C = 0.;
-                        L_rr_node[inode].set_rc_index(find_create_rr_rc_data(R, C));
-                        L_rr_node[inode].set_ptc_num(ipin);
-
-                        //Note that we store the grid tile location and side where the pin is located,
-                        //which greatly simplifies the drawing code
-                        L_rr_node[inode].set_coordinates(i + width_offset, j + height_offset, i + width_offset, j + height_offset);
-                        L_rr_node[inode].set_side(side);
-
-                        VTR_ASSERT(type->pinloc[width_offset][height_offset][L_rr_node[inode].side()][L_rr_node[inode].pin_num()]);
+                        if (OPEN != inode) {
+                            L_rr_node[inode].set_capacity(1);
+                            float R = 0.;
+                            float C = 0.;
+                            L_rr_node[inode].set_rc_index(find_create_rr_rc_data(R, C));
+                            L_rr_node[inode].set_ptc_num(ipin);
+
+                            //Note that we store the grid tile location and side where the pin is located,
```

Comment:
Should probably comment that if a pin is on multiple sides of a block or in multiple locations, we'll pick one of those sides / locations as its main location for drawing and other purposes.

Reply:
Indeed. Just added comments in the latest commit and hope it is clarified